server/src/main/java/org/jboss/as/server/logging/ServerLogger.java

```diff
@@ -1481,4 +1481,11 @@ public interface ServerLogger extends BasicLogger {
     @Message(id = Message.NONE, value = "%s started (with errors) in %dms - Started %d of %d services (%d services failed or missing dependencies, %d services are lazy, passive or on-demand) %s")
     String startedWitErrorsMessage(String prettyVersionString, long time, int startedServices, int allServices, int problemServices, int passiveOnDemandServices, String append);
 
+    @LogMessage(level = DEBUG)
+    @Message(id = Message.NONE, value= "Adding module dependency %s to module %s -- reason %s")
+    void logImplicitDependency(String dependency, String module, String reason);
+
+    @LogMessage(level = DEBUG)
+    @Message(id = Message.NONE, value= "Adding module dependency %s to module %s")
+    void logImplicitDependencyWithoutReason(String dependency, String module);
```

Comment:
The traces will be shown at DEBUG level and will contain a description provided by any DUPs without a proper translation. I would suggest just using the logger to print debug messages for debugging purposes.